Tests/PayID/PayIDClientTest.swift

```diff
@@ -0,0 +1,58 @@
+import Alamofire
+import Foundation
+import XCTest
+import XpringKit
+
+final class PayIDClientTest: XCTestCase {
+  func testAddressWithSuccessResponse() {
+    // GIVEN A PayIDClient with faked networking to return a successful response.
+    let address = "X7cBcY4bdTTzk3LHmrKAK6GyrirkXfLHGFxzke5zTmYMfw4"
+    let paymentNetwork = "xrpl-testnet"
+    let paymentInformation = PaymentInformation(
+      addresses: [
+        PayIdAddress(
+          paymentNetwork: paymentNetwork,
+          addressDetailsType: "",
+          addressDetails: CryptoAddressDetails(
+            address: address
+          )
+        )
+      ]
+    )
+
+    let jsonEncoder = JSONEncoder()
+    let response = try! jsonEncoder.encode(paymentInformation)
+
+    let manager: SessionManager = {
+      let configuration: URLSessionConfiguration = {
+        let configuration = URLSessionConfiguration.default
+        configuration.protocolClasses = [FakeURLProtocol.self]
+        return configuration
+      }()
+
+      return SessionManager(configuration: configuration)
+    }()
+
+    FakeURLProtocol.responseWithStatusCode(code: 200, asciiString: String(data: response, encoding: .utf8)!)
+
+    let payIDClient = PayIDClient(network: paymentNetwork, sessionManager: manager)
+
+    // WHEN the associated address is retrieved.
+    let expectation = XCTestExpectation(description: "Retrieved a PayID")
+    payIDClient.address(for: "georgewashington$xpring.money") { result in
+      // THEN the response contains the expected address.
+      switch result {
+      case .success(let addressDetails):
+        XCTAssertEqual(addressDetails.address, address)
+      case .failure(let error):
+        XCTFail("flagrant error: \(error)")
+      }
+
+      expectation.fulfill()
```

Comment:
If the network request is faked, then what exactly is being tested here? That creating a PayIDClient with a `xrpl-testnet` network doesn't throw an error?

Reply:
The parsing logic and error handling logic is also tested. 

Admittedly, this is a pretty thin test. In the future we will have cases for:
- Malformed response
- Not found
- Generic network error

Reply:
> The parsing logic and error handling logic is also tested.
>
> Admittedly, this is a pretty thin test. In the future we will have cases for:
>
> Malformed response
> Not found
> Generic network error

Got it. Is there a reason to prefer mocking network requests to test this stuff vs writing unit tests that cover the same areas? Is it that you want the E2E nature of a network test?

Reply:
In this case, I prefer mocking because I'm relying heavily on Swagger and not really using it as intended. I could certainly find a way to inject a swagger API client and verify that it works. 

This was originally a holdover from when we started doing just plain network requests (before OpenAPI code generation). Basically, the reason we moved to code generation was because of the schemas in PayID+ being more difficult. 

I don't really have an opinion here. I can always find a way to fake these things; in this particular instance I think mocking might give us a little more flexibility if we ever change the class implementation. I also see the argument that we should be consistent in our test strategy though. 

Overall, I'm motivated to try to land this as-is rather than re-writing as a fake but you can LMK if you feel strongly?

Reply:
No, I think network mocking makes sense here, if the other option is faking.

I'm just intuitively skeptical of mocks/fakes in tests, because then you need to think a little harder about what you're actually testing, but you've clearly thought that through, so it's fine.